VideoLocker/src/main/java/org/edx/mobile/loader/CoursesAsyncLoader.java

```diff
@@ -52,6 +45,18 @@ public CoursesAsyncLoader(Context context) {
         return result;
     }
 
+    private List<EnrolledCoursesResponse> getEnrolledCourses() {
+        List<EnrolledCoursesResponse> ret = null;
+
+        try {
+            ret = environment.getServiceManager().getEnrolledCourses(false);
```

Comment:
> Retrofit may have their own cache

OkHttp handles the HTTP layer, and likewise any HTTP caching. Retrofit has no cache of its own.

> Yeah, a challenge that I ran into while doing this was just figuring out what APIs I should use and why.

Backstory: We are in the process of switching from an old deprecated HTTP client (ApacheHttpClient) to a more modern one (OkHttp).

All new code should go through OkHttp. We have been using Retrofit's interface (backed by OkHttp) to define our endpoints rather than working with OkHttp directly. (It's debatable whether that is helpful or harmful overall, due to the inflexibility of defining everything via `@Annotations` and not code.)